### PR TITLE
Added sklearn classification pipeline to main workflow.

### PR DESCRIPTION
**Pull request by dickeygh**
Created new file called `Classifiers.py`, works similarly to `regressors.py`.

Added a new function called classification_metrics (line 316) to `analysis.py`. This function computes and returns several different evaluation metrics of the model's performance on the test set.

Added classification algorithms svc and kneighborsclassifier to the list of models that need feature normalization in `features.py` (line 23).

Changed `main.py` to run either the classification models or the regression models given a user input.

Changed line 17 of `name.py` to fit my directory.

Added a classification_run function to `models.py` that runs the classification models and prints evaluation results.

Changed `ingest.py` line 4 so that drop is not always=true. This is because some of the classification models run better on some data sets without dropping the extra columns. (I made drop=True when running regression models, see `main.py` line 64.)

Added sider data set to dataFiles.

Currently have 4 evaluation methods (accuracy score, confusion matrix, roc auc score, and classification report), 1 data set (sider), and 2 different models (SVC and Kneighborsclassifier). I plan on adding more of all of these.

### CODE REVIEW

File: core/features.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def featurize(df, model_name, num_feat=None):
                  'morganchiral3counts', 'atompaircounts']
 
     # Remove un-normalized feature option depending on model type
-    if model_name in ['knn', 'mlp', 'svr']:
+    if model_name in ['knn', 'mlp', 'svr', 'svc', 'knc']:
```
Comment (qle2) [inline core/features.py:23]: This is just a question that does not relate to your code: Have you checked how machine learning results differ between different normalization methods available in sklearn for your classification models?

File: core/Classifiers.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from sklearn.svm import SVC
```
Comment (qle2) [inline core/Classifiers.py:1]: Nothing wrong with the code. I suggest renaming your script to `classifiers.py`. Just a quality-of-live kind of improvement.

File: main.py [hunk 3]
```diff
@@ -12,66 +12,90 @@ def main():
...
-    }
-
-    for alg in learner:  # loop over all learning algorithms
-
-        if alg in ['mlp', 'knn', 'svr']: # if the algorithm needs normalized data
-            feats = norm_featurize
-        else:
-            feats = featurize
+    # Asking the user to decide between running classification models or regression models
+    C = str(input("Enter classification for classification, regression for regression: "))
```
Comment (qle2) [inline main.py:16]: Generally we don't use upper case to name variables. Same reason as naming files. 

Also, typing `classification` and `regression` is very long and we are bound to make typos. Use other strings instead like `c` for classification and `r` for regression.

**Comment by qle2** [comment]
It's well put together overall. Just some minor naming convention stuff (This guide is pretty helpful: https://www.python.org/dev/peps/pep-0008/#naming-conventions). I like how well documented your functions are scripts are.

File: core/Classifiers.py (hunk 2 above)
Comment (Dlux804) [reply core/Classifiers.py:1]: I agree with Quang.  Use the lower case, please.

File: core/features.py (hunk 1 above)
Comment (dickeygh) [reply core/features.py]: I have done some research into this and I have also found that these models generally perform better with the rdkit2dnormalized featurization option (some won't run properly with rdkit2d).

**Comment by dickeygh** [comment]
I made the following changes to this pull request on 6/3:
Moved importing of classification evaluation metrics to the top of the analysis.py file.

Added an extra classifier (RandomForestClassifier) to classifiers.py. This also resulted in small changes on line 23 of features.py and line 21 of main.py.

Changed classifiers.py to be lowercase.

Changed the variable "c" to be lowercase in main.py. Changed the input statement to prompt for c and r instead of classification and regression. (Line 16)

Changed line 17 of name.py to work across different machines.

File: core/models.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 This code was written by Adam Luxon and team as part of the McQuade research group.
 '''
 
-from core import ingest, features, grid, regressors, analysis, name
+from core import ingest, features, grid, regressors, analysis, name, classifiers
```
Comment (qle2) [inline core/models.py:5]: Random Forest has a property called `feature_importances_` that will give a percentage score to all features on how important they are or how much they contribute to the learning process. We use that property to make feature importance graph in line 81 and 82. Can you please add in the string representation of Random Forest Classification on the _if_ statement in line 81?
